exn/src/impls.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright 2025 tison <wander4096@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::marker::PhantomData;
+use std::panic::Location;
+
+use crate::Error;
+
+/// An exception type that can hold an error tree and additional context.
+pub struct Exn<E> {
+    // trade one more indirection for less stack size
+    frame: Box<ExnFrame>,
+    // E is invariant
+    invariant: PhantomData<E>,
```

Comment:
`PhantomData<E>` actually is actually covariant. The reason why `Exn<E>` is invariant is that E is `'static`, and `PhantomData<E>` here is for type erasing, so I think a common `_phantom` is appropriate.

Reply:
Then let's call it `covariant`.

I'd prefer more expressive var name and would avoid leading `_` since we now have `#[expect]`

Reply:
Get it. Then it's `phantom` that only make rustc happy😇